crates/aranya-fast-channels/src/shm/align.rs

```diff
@@ -12,12 +12,18 @@ use super::shared::assert_ffi_safe;
 /// Reports whether `v` is aligned to `align`.
 pub(super) fn is_aligned_to(v: usize, align: usize) -> bool {
     assert!(align.is_power_of_two());
-    v % align == 0
+    #[allow(clippy::arithmetic_side_effects)] // checking alignment
+    {
+        v % align == 0
+    }
 }
 
 /// Reports whether `ptr` is aligned to `align`.
 fn is_aligned<T>(ptr: *const T) -> bool {
-    (ptr as usize) % align_of::<T>() == 0
+    #[allow(clippy::arithmetic_side_effects)] // checking pointer alignment
```

Comment:
`// checking pointer alignment` is also a bad comment. The real reason is that `align_of::<T>()` is always non-zero. Also, use `#[allow(..., reason = "...")]` to explain why.